scripts/helpers.py

```diff
@@ -0,0 +1,152 @@
+#
+# helpers
+#
+
+import codecs
+import functools
+import json
+import multiprocessing
+import os
+import re
+import subprocess
+import sys
+from urllib.request import Request, urlopen, urlretrieve
+
+
+def _camel_hyphen(string):
+    return re.sub(r'([a-z])([A-Z])', r'\1-\2', string)
+
+
+def _load_image(cache_dir, url):
+    filename = os.path.basename(url)
+    filepath = os.path.join(cache_dir, filename)
+    etag_cache_file = filepath + '.etag'
+    if os.path.exists(etag_cache_file):
+        with open(etag_cache_file, mode='r') as f:
+            etag = f.read().strip()
+    else:
+        etag = None
+    request = Request(url)
+    request.get_method = lambda: 'HEAD'
+    try:
+        response = urlopen(request)
+    except Exception as e:
+        print('Error while fetching %s: %s' % (url, str(e)))
+        return False
+    new_etag = response.info().get('ETag')
+    if etag == new_etag:
+        print("Skipping download of %s, local file is current" % filename)
+        return True
+    print("downloading", url)
+    try:
+        os.makedirs(cache_dir)
+    except Exception:  # noqa: E722
+        pass  # ignore
+    try:
+        urlretrieve(url, filepath)
+    except Exception as e:
+        print('Error while fetching %s: %s' % (url, str(e)))
+        return False
+    subprocess.check_call(["docker", "load", "-i", filepath])
+    with open(etag_cache_file, mode='w') as f:
+        f.write(new_etag)
+    return True
+
+
+def load_images(urls, cache_dir):
+    load_image_fn = functools.partial(_load_image, cache_dir)
+    pool = multiprocessing.Pool(4)
+    # b/c python2
+    try:
+        results = pool.map_async(load_image_fn, urls).get(timeout=10000000)
+    except KeyboardInterrupt:
+        pool.terminate()
+        raise
+    if not all(results):
+        print("Errors while downloading. Exiting.")
+        sys.exit(1)
+
+
+DEFAULT_HEALTHCHECK_INTERVAL = "10s"
+DEFAULT_HEALTHCHECK_RETRIES = 12
+
+
+def curl_healthcheck(port, host="localhost", path="/healthcheck",
+                     interval=DEFAULT_HEALTHCHECK_INTERVAL, retries=DEFAULT_HEALTHCHECK_RETRIES):
+    return {
+        "interval": interval,
+        "retries": retries,
+        "test": ["CMD", "curl", "--write-out", "'HTTP %{http_code}'", "--fail", "--silent",
+                 "--output", "/dev/null",
+                 "http://{}:{}{}".format(host, port, path)]
+    }
+
+
+build_manifests = {}  # version -> manifest cache
```

Comment:
Uhm, should it be moved to the method itself?

Reply:
@v1v `build_manifests` acts as a global cache here, so subsequent calls to `resolve_bc` with the same arguments don't have to make expensive network requests.

Since we're switching to Python 3, we could theoretically use [`functools.lru_cache`](https://docs.python.org/3/library/functools.html#functools.lru_cache) which achieves the same in a less hacky way, but let's leave that for after this refactor.

Reply:
Thanks :)